indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractBatchIndexTask.java

```diff
@@ -704,7 +704,8 @@ protected static List<DataSegment> findInputSegments(
   protected boolean waitForSegmentAvailability(
       TaskToolbox toolbox,
       List<DataSegment> segmentsToWaitFor,
-      long waitTimeout
+      long waitTimeout,
+      String taskId
```

Comment:
@hardikbajaj , I just realized, you need not pass the `taskId` as argument here. See the other comment.